### PR TITLE
ft: ZENKO-1346 update ingestion object metadata

### PR DESCRIPTION
**Pull request by philipyoo**
sits on top of https://github.com/scality/backbeat/pull/580

### CODE REVIEW

File: extensions/mongoProcessor/MongoQueueProcessor.js [hunk 1]
```diff
@@ -221,25 +299,53 @@ class MongoQueueProcessor {
...
+                return process.nextTick(done);
+            }
+
             if (sourceEntry instanceof DeleteOpQueueEntry) {
                 return this._processDeleteOpQueueEntry(sourceEntry, done);
             }
             if (sourceEntry instanceof ObjectQueueEntry) {
-                return this._processObjectQueueEntry(sourceEntry, ownerInfo,
-                    done);
+                return this._processObjectQueueEntry(sourceEntry, location,
```
Comment (JianqinWang) [inline extensions/mongoProcessor/MongoQueueProcessor.js:344]: https://github.com/scality/backbeat/blob/eb56ee104e2ced8ca1c4d10c1b1ea26c2a33c976/extensions/mongoProcessor/MongoQueueProcessor.js#L159

only has 3 params taken?

File: extensions/mongoProcessor/MongoQueueProcessor.js [hunk 2]
```diff
@@ -221,25 +296,53 @@ class MongoQueueProcessor {
         }
 
         const bucketName = sourceEntry.getBucket();
-        return this._mongoClient.getBucketAttributes(bucketName, this.logger,
-        (err, data) => {
+        return async.series([
```
Comment (JianqinWang) [inline extensions/mongoProcessor/MongoQueueProcessor.js:299]: not a big deal but this could be `async.parallel` instead of `async.series`

**Review by jonathan-gramain — APPROVED** [review]
I assumed the `_updateLocation()` and related functions are to map the MetaData location names to zenko location names. Maybe this could be documented at a high level, either in the class jsdoc or a separate README file explaining the general idea.

Also, not related to this PR, but the module is called "ingestion" in the config description but is found in the "mongoProcessor" directory, it could be confusing.

**Comment by rahulreddy** [comment]
Yeah I have some peeves about naming too. I think we are going to address it with a couple of cleanup PRs.

**Review by rahulreddy — APPROVED** [review]

**Comment by rahulreddy** [comment]
@bert-e help

**Comment by rahulreddy** [comment]
@bert-e bypass_author_approval

**Pull request merged**